test/evergreen/code_change_report/code_change_info.py

```diff
@@ -244,21 +244,17 @@ def get_function_info(file_path: str,
                 if preprocessed_prev_complexity_data is not None:
                     file_prev_info = preprocessed_prev_complexity_data[detail_file]
                     if function_name in file_prev_info:
-                        function__prev_info = file_prev_info[function_name]
-                        function_info['prev_complexity'] = int(function__prev_info['std.code.complexity:cyclomatic'])
-                        function_info['prev_lines_of_code'] = int(function__prev_info['std.code.lines:code'])
+                        function_prev_info = file_prev_info[function_name]
+                        function_info['prev_complexity'] = int(function_prev_info['std.code.complexity:cyclomatic'])
+                        function_info['prev_lines_of_code'] = int(function_prev_info['std.code.lines:code'])
                 break
 
     return function_info
 
 
 # get_num_branches_covered returns the number of branches covered from coverage info
 def get_num_branches_covered(branch_coverage_info: list) -> int:
-    num_branches_covered = 0
-    for branch in branch_coverage_info:
-        if branch['count'] > 0:
-            num_branches_covered += 1
-    return num_branches_covered
+    return len([x for x in branch_coverage_info if x['count'] > 0])
```

Comment:
This is a typo from my original suggestion. The `> 0` needs to be outside the `len` call:
```suggestion
    return len([x for x in branch_coverage_info if x['count']])
```
```suggestion
    return len([x for x in branch_coverage_info if x['count'] > 0]) > 0
```

Reply:
Hi @ajmorton , I believe the original code is correct (ie with the `>0` inside the `len` call) because we want to count the number of branches that have an execution count greater than 0.
I've added an extra comment and split the code up to (hopefully) make it easier to understand.

Reply:
Ah yep my bad. I had it in my head we were returning a bool not an int.
Resolving